src/components/Portal/PortalConsumer.tsx

```diff
@@ -23,11 +23,15 @@ export default class PortalConsumer extends React.Component<Props> {
   componentDidUpdate() {
     this.checkManager();
 
-    this.props.manager.update(
-      this.key,
-      this.props.children,
-      this.props.isFocused
-    );
+    // Because of the delay in componentDidMount, componentDidUpdate
+    // can run before a key is set causing corrupt data in the manager.
+    if (this.key) {
```

Comment:
From what I can tell, `componentDidUpdate` is not guaranteed to run after the key is set (eg when `this.props.manager.mount` completes). Mount will update the `PortalManager.state` (component inside `PortalHost`), but doesn't necessarily mean that the `PortalConsumer` will be updated.
If there's a race condition where the component can mount and receive new props before `componentDidMount` completes, those props would not be updated in the `PortalManager` I think. Is there a better solution than the async behaviour in `componentDidMount`?